packages/medusa/src/commands/plugin/develop.ts

```diff
@@ -7,13 +8,43 @@ export default async function developPlugin({
 }: {
   directory: string
 }) {
+  let isBusy = false
   const compiler = new Compiler(directory, logger)
+  const yalcBin = path.join(path.dirname(require.resolve("yalc")), "yalc.js")
+
   await compiler.developPluginBackend(async () => {
-    await yalc.publishPackage({
-      push: true,
-      workingDir: directory,
-      changed: true,
-      replace: true,
-    })
+    /**
+     * Here we avoid multiple publish calls when the filesystem is
+     * changed too quickly. This might result in stale content in
+     * some edge cases. However, not preventing multiple publishes
+     * at the same time will result in race conditions and the old
+     * output might appear in the published package.
+     */
+    if (isBusy) {
+      return
+    }
+    isBusy = true
+
+    /**
+     * Yalc is meant to be used a binary and not as a long-lived
+     * module import. Therefore we will have to execute it like
+     * a command to get desired outcome. Otherwise, yalc behaves
+     * flaky.
+     */
+    execFile(
+      yalcBin,
+      ["publish", "--push", "--no-scripts"],
+      {
+        cwd: directory,
+      },
+      (error, stdout, stderr) => {
+        isBusy = false
+        if (error) {
+          console.log(error)
```

Comment:
Q: Could we use the logger instead?

Reply:
This is the error from the child process, which might already be formatted in a certain way. So I think logging it as it is makes more sense. But yes, I can use `console.error` instead.